### PR TITLE
Fix GPU fault issue when run dsv3 with dp mode and enable torch-compile

### PR DESCRIPTION
**Pull request by kkHuang-amd**
## Motivation

Got GPU fault when running "python3 -m sglang.launch_server --model-path DeepSeek-V3.1 --tp 8 --trust-remote-code --chunked-prefill-size 131072 --dp-size 8 --enable-dp-attention --enable-torch-compile"

## Modifications

The cored-dump happened is caused by using class member in the different places. 
Need to ensure the data read will not affect by other thread or process.

Change the code of logitprocess as independent region.

## Accuracy Tests
image

## Benchmarking and Profiling
**[Before this commit]**
image

**[after this commit]**
image

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.ai/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.ai/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.ai/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.ai/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.ai/developer_guide/contribution_guide.html#benchmark-the-speed).

### CODE REVIEW

**Comment by HaiShaw** [comment]
It is about `set_dp_buffer_len` being used from both `cuda_graph_runner.py` and `logits_processor.py`, causing races seen on rocm.
@kkHuang-amd Please comment more.

**Review by HaiShaw — APPROVED** [review]

**Comment by HaiShaw** [comment]
@kkHuang-amd provided explanation in `Modification`.
